evidence.go

```diff
@@ -194,17 +218,22 @@ func (e *Evidence) SetUnvalidatedClaims(p platform.IClaims, r realm.IClaims) err
 	if p != nil && r != nil {
 		_ = e.bind()
 	}
+}
 
-	return nil
+// ValidateAndSign validates and then signs the given evidence using the
```

Comment:
In my view, we should not mirror Signing to look exactly as CBOR and JSON Marshal/UnMarshal.

A Sign method, should always Validate! 
This is the current `mainline` `implementation`

If needed ( I am still struggling with the use case though), we should provide a new method:
that lets caller, Sign UnValidatedClaims().

Reply:
Sigining and validation are two distinct operations that if the calling code wants to perform separately, it should . E.g. when generationg knowingly bad inputs (such as for testing, or to work around quirks), or to avoid annecessary  re-validation when signing a previously validated structure.

Reply:
If you see the commit message, this change, does not align well with the commit intentions and in general the scope of the PR. I suggest, we can discuss this internally and once everyone agrees then re-insert this part of change which modifies the existing sign interface, a s separate PR!

Reply:
Good point about the commit message; i reused the `pastoken` one, and it should be ammended to reflect this change as well. However, this change is absolutely aligned with the intent behind this commit, which is to align with convenetion/expectation and reduce surprise. And is absolutely not the sepectation that `Sign()` performs validation as well (especially as there is already a separate `Validate()` ).

While sepate from marshaling, `Sign()` is part of the same problem, namely that validation has been confounded with other operations, necessaitating the addition of the cruft-y "unvalidated" API, and fixing the problem involves fixing `Sign()` as well.

Reply:
ok, then please change the commit title to Marshaling and Sign refactor, to be clear. 

Also make the same change to `psatoken` as it is symmetrical